src/plugins/embeddable/public/react_embeddable_system/types.ts

```diff
@@ -19,26 +19,34 @@ export type ReactEmbeddableRegistration<
  *
  * Before adding anything to this interface, please be certain that it belongs in *every* embeddable.
  */
-export type DefaultEmbeddableApi = DefaultPresentationPanelApi &
-  PublishesUnsavedChanges & {
-    serializeState: () => Promise<SerializedPanelState>;
-  };
+export interface DefaultEmbeddableApi<StateType extends object = object>
+  extends DefaultPresentationPanelApi,
+    HasType,
+    PublishesUnsavedChanges {
+  serializeState: () => SerializedPanelState<StateType>;
```

Comment:
Do you want to move `serializeState` definition into its own interface in presentation-containers package?

Reply:
Sure, I've addressed this in 8d46fc32103